File: src/main/java/com/extendedae_plus/integration/ContextModLoaded.java
package com.extendedae_plus.integration;

import com.extendedae_plus.common.init.InitObject;
import lombok.Getter;
import net.neoforged.fml.ModList;
import org.jetbrains.annotations.Nullable;

public enum ContextModLoaded {
    emi("emi"),
    jei("jei"),
    ftbTeams("ftbteams"),
    jech("jecharacters"),
    curios("curios"),
    ae2wtlib("ae2wtlib"),
    advancedAE("advanced_ae"),
    appliedFlux("appflux"),
    mekanism("mekanism"),
    appliedMekanistics("appmek"),
    gtceuModern("gtceu"),
    expandedAE("expandedae", false),

    ;

    private final String modID;
    @Getter
    private boolean loaded;
    private final @Nullable Boolean required;

    ContextModLoaded(String modID) {
        this(modID, null);
    }

    ContextModLoaded(String modID, @Nullable Boolean required) {
        this.modID = modID;
        this.loaded = false;
        this.required = required;
    }

    boolean shouldTip() {
        return this.required != null && this.required != this.loaded;
    }

    @InitObject(priority = 0)
    private static void init() {
        for (ContextModLoaded context : ContextModLoaded.values()) {
            context.loaded = ModList.get().isLoaded(context.modID);
        }
    }
}

File: src/main/kotlin/com/extendedae_plus/integration/ContextModLoaded.kt
package com.extendedae_plus.integration

import com.extendedae_plus.common.init.InitObject
import lombok.Getter
import net.neoforged.fml.ModList

enum class ContextModLoaded(
    private val modID: String,
    private val required: Boolean? = null
) {
    Emi("emi"),
    Jei("jei"),
    FTBTeams("ftbteams"),
    Jech("jecharacters"),
    Curios("curios"),
    AE2wtlib("ae2wtlib"),
    AdvancedAE("advanced_ae"),
    AppliedFlux("appflux"),
    Mekanism("mekanism"),
    AppliedMekanistics("appmek"),
    GtceuModern("gtceu"),
    ExpandedAE("expandedae", false),
    ;

    @Getter
    private var loaded = false

    fun shouldTip() = this.required != null && this.required != this.loaded

    companion object {
        @InitObject(priority = 0)
        private fun init() {
            for (context in entries) {
                context.loaded = ModList.get().isLoaded(context.modID)
            }
        }
    }
}
